modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheEntryEx.java

```diff
@@ -1110,9 +1110,28 @@ public void updateIndex(
      *
      * @param ver Version.
      * @param ttl Time to live.
+     * @return Cached object.
      * @throws GridCacheEntryRemovedException If entry is obsolete (entry was removed).
      */
-    public void updateTimeToLiveOnTtlUpdateRequest(@Nullable GridCacheVersion ver, long ttl) throws GridCacheEntryRemovedException;
+    public CacheObject updateTimeToLiveOnTtlUpdateRequest(@Nullable GridCacheVersion ver, long ttl) throws GridCacheEntryRemovedException;
+
+    /**
+     * Updates ttl for this entry and stores its value in storage.
+     *
+     * @param expiryPlc Expiry policy to be uased to calculate new ttl value.
+     * @return Cached object.
+     * @throws GridCacheEntryRemovedException If entry is obsolete (entry was removed).
+     */
+    public CacheObject touchTtl(@Nullable IgniteCacheExpiryPolicy expiryPlc) throws GridCacheEntryRemovedException;
+
+    /**
+     * Updates ttl for this entry and stores its value in storage.
+     *
+     * @param expiryPlc Expiry policy to be uased to calculate new ttl value.
+     * @return Cached object.
+     * @throws GridCacheEntryRemovedException If entry is obsolete (entry was removed).
+     */
+    public EntryGetResult touchTtlVersioned(@Nullable IgniteCacheExpiryPolicy expiryPlc) throws GridCacheEntryRemovedException;
```

Comment:
It's unclear from the javadoc and from the method names what is the difference between them and what are their use cases.